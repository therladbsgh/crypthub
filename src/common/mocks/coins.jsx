export const coins = [
    {
        name: 'Bitcoin',
        symbol: 'BTC',
        currPrice: 10000
    },
    {
        name: 'Ethereum',
        symbol: 'ETH',
        currPrice: 5000
    }
]
