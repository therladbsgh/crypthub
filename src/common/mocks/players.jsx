import date from 'date-and-time';

export const player1 = {
    name: 'player1',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 250
        },
        {
            name: 'Ethereum',
            amount: 1000.28373623
        },
        {
            name: 'USD',
            amount: 250
        }
    ],
    transactions: {
        history: [
            {
                id: '1',
                type: 'market',
                side: 'buy',
                size: 25,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 1 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: true,
                filledDate: date.parse('Apr 3 2018', 'MMM D YYYY')
            },
            {
                id: '2',
                type: 'limit',
                side: 'buy',
                size: 35.000006,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('Apr 5 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ],
        current: [
            {
                id: '3',
                type: 'short',
                side: 'sell',
                size: 80,
                price: 100,
                symbol: 'BTC',
                date: date.parse('Apr 2 2018', 'MMM D YYYY'),
                GTC: true,
                expiration: null,
                filled: false,
                filledDate: null
            },
            {
                id: '4',
                type: 'limit',
                side: 'buy',
                size: 2,
                price: 200,
                symbol: 'ETH',
                date: date.parse('Apr 7 2018', 'MMM D YYYY'),
                GTC: false,
                expiration: date.parse('May 2 2018', 'MMM D YYYY'),
                filled: false,
                filledDate: null
            }
        ]
    },
    netWorth: 200,
    numTrades: 80,
    netReturn: -20,
    todayReturn: 5,
    currRank: 2,
    buyingPower: 150,
    shortReserve: 50
};

export const player2 = {
    name: 'player2',
    portfolio: [
        {
            name: 'Bitcoin',
            amount: 300
        }
    ],
    transactions: {
        history: [
            {
                id: '5',
                side: 'buy',
                size: 1,
                price: 10,
                symbol: 'ETH'
            },
            {
                id: '6',
                side: 'sell',
                size: 3500,
                price: 202220,
                symbol: 'ETH'
            }
        ],
        current: [
            {
                id: '7',
                side: 'sell',
                size: 23230,
                price: 232,
                symbol: 'BTC'
            },
            {
                id: '8',
                side: 'buy',
                size: 6000,
                price: 50,
                symbol: 'ETH'
            }
        ]
    },
    netWorth: 300,
    numTrades: 2,
    netReturn: 120
};
