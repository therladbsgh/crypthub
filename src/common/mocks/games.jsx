import date from 'date-and-time';
import * as PlayerMocks from './players';

export const game1 = {
    id: '1',
    name: 'game1',
    description: 'desc1',
    host: 'user1',
    created: date.parse('Apr 1 2018', 'MMM D YYYY'),
    start: date.parse('Apr 2 2018', 'MMM D YYYY'),
    end: date.parse('May 7 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game2 = {
    id: '2',
    name: 'game2',
    description: 'desc2',
    host: 'user2',
    created: date.parse('Apr 2 2018', 'MMM D YYYY'),
    start: date.parse('Apr 3 2018', 'MMM D YYYY'),
    end: date.parse('May 1 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};

export const game3 = {
    id: '3',
    name: 'game3',
    description: 'desc3',
    host: 'user3',
    created: date.parse('Apr 4 2018', 'MMM D YYYY'),
    start: date.parse('Apr 8 2018', 'MMM D YYYY'),
    end: date.parse('Apr 28 2018', 'MMM D YYYY'),
    playerPortfolioPublic: true,
    startingBalance: 10000,
    commissionValue: 10,
    shortSelling: true,
    limitOrders: true,
    stopOrders: true,
    lastUpdated: new Date(),
    completed: false,
    players: [
        PlayerMocks.player1,
        PlayerMocks.player2
    ],
    isPrivate: false,
    password: ''
};
