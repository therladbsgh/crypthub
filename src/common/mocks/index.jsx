import * as GameMocks from './games';
import * as PlayerMocks from './players';

export { GameMocks, PlayerMocks };
