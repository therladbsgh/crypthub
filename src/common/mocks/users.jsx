import * as GameMocks from './games';

export const user1 = {
    id: 'id1',
    name: 'user1',
    ELO: 3000,
    games: [
        GameMocks.game1,
        GameMocks.game2
    ]
};

export const user2 = {
    id: 'id2',
    name: 'user2',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};

export const user3 = {
    id: 'id3',
    name: 'user3',
    ELO: 2000,
    games: [
        GameMocks.game2,
        GameMocks.game3
    ]
};
