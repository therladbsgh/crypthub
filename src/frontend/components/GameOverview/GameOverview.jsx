import * as _ from 'lodash';
import React, { Component } from 'react';
import { Tab, Header, Dropdown } from 'semantic-ui-react';
import { Searchbar, ProfileCard, GameCard } from 'components';

export default class GameOverview extends Component {
    render() {
        const { game, thisPlayer, completed, inGame } = this.props;

        return (
			<div>
				{inGame && 
                [<Header key='1' as='h2'>Your Profile</Header>,
                <ProfileCard key='2' player={thisPlayer} completed={completed} />]}
				<Header as='h2'>About This Game</Header>
                <GameCard game={game} />
			</div>
        );
    }
}
