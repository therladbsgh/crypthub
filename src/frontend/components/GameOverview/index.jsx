import GameOverview from './GameOverview';

export default GameOverview;
