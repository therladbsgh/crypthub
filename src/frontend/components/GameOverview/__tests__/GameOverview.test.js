import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import GameOverview from '../GameOverview';
import { GameMocks } from 'mocks';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  	const tree = renderer.create(<Router><GameOverview game={GameMocks.game1} /></Router>).toJSON();
  	expect(tree).toMatchSnapshot();
});