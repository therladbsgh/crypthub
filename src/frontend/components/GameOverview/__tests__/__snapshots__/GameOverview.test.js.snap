// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <h2
    className="ui header"
  >
    About This Game
  </h2>
  <div
    className="ui card"
    href={undefined}
    id={undefined}
    onClick={[Function]}
  >
    <div
      className="content"
    >
      <div
        className="header"
      >
        <a
          href="/game/1"
          onClick={[Function]}
        >
          game1
        </a>
        <span
          className="false undefined"
        >
          
        </span>
      </div>
      <div
        className="meta"
      >
        desc1
      </div>
      <div
        className="description"
      >
        Created 
        <strong>
          26 days
           ago
        </strong>
         by 
        <strong>
          user1
        </strong>
         to start on 
        <strong>
          04/02/2018
        </strong>
         and end 
        in
         
        <strong>
          9 days
        </strong>
      </div>
    </div>
    <div
      className="extra content"
    >
      2
       players
    </div>
  </div>
</div>
`;
